config.go

```diff
@@ -338,3 +338,31 @@ func nodejs(c *Config) error {
 
 	return nil
 }
+
+// python config.
+func python(c *Config) {
+	if c.Proxy.Command == "" {
+		c.Proxy.Command = "python app.py"
+	}
+
+	// Only add build & clean hooks if a requiremnts.txt exists
+	if !util.Exists("requirements.txt") {
+		return
+	}
+
+	// Set PYTHONPATH env
+	if c.Environment == nil {
+		c.Environment = config.Environment{}
+	}
+	c.Environment["PYTHONPATH"] = ".pypath/"
+
+	// Copy libraries into .pypath/
+	if c.Hooks.Build == "" {
+		c.Hooks.Build = `mkdir -p .pypath/ && pip install -r requirements.txt -t .pypath/`
```

Comment:
It is suboptimal / "expensive" since it does fetch all the deps on each `build` / `clean` cycle (the `flask` and `bottle` examples take 2 to 5s to build locally).

Unfortunately Python has nothing comparable to `node_modules/` or `vendor/`, all dependencies are global by default. In practice people use `virtualenv`s to essentially switch between environments (think switching `$GOPATH`s).

AWS' [docs suggest](http://docs.aws.amazon.com/lambda/latest/dg/lambda-python-how-to-create-deployment-package.html) copying all of `site-packages/*` into your root folder. That isn't ideal since it would pollute the current project's folder (and would be impossible to clean), which is why I opted to isolate it under `./.pypath/`.

---

We could make the `build` faster by simply copying the global `site-packages/*` at the cost of a larger lambda bundle (since it could copy unrelated/unused libraries installed in the global path) or by adding some smart python specific logic.

I opted for a first solution that requires little new code/logic and produces decent bundles at the expense of longer build times.